src/System.Collections/tests/BitArray/BitArray_OperatorsTests.cs

```diff
@@ -2,140 +2,153 @@
-            }
+            BitArray bitArray = new BitArray(data);
 
             BitArray bitArrayNot = bitArray.Not();
             Assert.Equal(bitArray.Length, bitArrayNot.Length);
             Assert.Same(bitArray, bitArrayNot);
             for (int i = 0; i < bitArray.Length; i++)
             {
-                if (i <= 1 || i >= length - 2)
-                {
-                    Assert.False(bitArrayNot[i]);
-                }
-                else
-                {
-                    Assert.True(bitArrayNot[i]);
-                }
+                Assert.Equal(!data[i], bitArrayNot[i]);
+            }
+        }
+
+        public static IEnumerable<object[]> And_Operator_Data()
+        {
+            yield return new object[] { new bool[0], new bool[0], new bool[0] };
+            foreach (int size in new[] { 0, 1, BitsPerByte, BitsPerByte * 2, BitsPerInt32, BitsPerInt32 * 2, short.MaxValue })
+            {
+                bool[] allTrue = Enumerable.Repeat(true, size).ToArray();
+                bool[] allFalse = Enumerable.Repeat(false, size).ToArray();
+                bool[] alternating = Enumerable.Range(0, size).Select(i => i % 2 == 1).ToArray();
+                yield return new object[] { allTrue, allTrue, allTrue };
+                yield return new object[] { allTrue, allFalse, allFalse };
+                yield return new object[] { allFalse, allTrue, allFalse };
+                yield return new object[] { allFalse, allFalse, allFalse };
+                yield return new object[] { allTrue, alternating, alternating };
+                yield return new object[] { alternating, allTrue, alternating };
+                yield return new object[] { allFalse, alternating, allFalse };
+                yield return new object[] { alternating, allFalse, allFalse };
             }
         }
-        
+
         [Theory]
-        [InlineData(Operator.And)]
-        [InlineData(Operator.Or)]
-        [InlineData(Operator.Xor)]
-        public static void OperatorTest(Operator op)
+        [MemberData(nameof(And_Operator_Data))]
+        public static void And_Operator(bool[] l, bool[] r, bool[] expected)
         {
-            BitArray bitArray1 = new BitArray(6, false);
-            BitArray bitArray2 = new BitArray(6, false);
-            BitArray result;
+            BitArray left = new BitArray(l);
+            BitArray right = new BitArray(r);
+
+            BitArray actual = left.And(right);
+            Assert.Same(left, actual);
+            Assert.Equal(actual.Length, expected.Length);
+
+            for (int i = 0; i < expected.Length; i++)
+            {
+                Assert.Equal(expected[i], actual[i]);
+            }
+        }
+
+        public static IEnumerable<object[]> Or_Operator_Data()
+        {
+            yield return new object[] { new bool[0], new bool[0], new bool[0] };
+            foreach (int size in new[] { 0, 1, BitsPerByte, BitsPerByte * 2, BitsPerInt32, BitsPerInt32 * 2, short.MaxValue })
```

Comment:
Why 0?  Doesn't that make all eight of the below cases identical to the 0,0,0 case above?